File: iOS/HUMAN_SDK_Demo/HUMAN_SDK_Demo/HUMAN/HumanManager.swift
//
//  HumanManager.swift
//  HUMAN_SDK_Demo
//
//  Created by HUMAN.
//

import Foundation
import HUMAN

class HumanManager: NSObject, HSBotDefenderDelegate {
    
    // MARK: - HSBotDefenderDelegate
    
    func botDefenderRequestBlocked(url: URL?, appId: String) {
        print("Request Blocked")
    }
    
    func botDefenderChallengeSolved(forAppId appId: String) {
        print("Challenge Solved")
    }
    
    func botDefenderChallengeCancelled(forAppId appId: String) {
        print("Challenge Cancelled")
    }
    
    func botDefenderChallengeRendered(forAppId appId: String) {
        print("Challenge Rendered")
    }
    
    func botDefenderChallengeRenderFailed(forAppId appId: String) {
        print("Challenge Render Failed")
    }
    
    func botDefenderDidUpdateHeaders(headers: [String : String], forAppId appId: String) {
        print("Headers Were Updated")
    }
    
    // MARK: - singleton
    
    static let shared = HumanManager()
    
    // MARK: - properties
    
    private let pxAppId = "PXj9y4Q8Em"
    let urlRequestInterceptionType: HSAutomaticInterceptorType = .interceptWithDelayedResponse
    
    // MARK: - HUMAN
    
    func start() {
        print("SDK version: \(HumanSecurity.sdkVersion())")
        
        start(appId: pxAppId)
        setCustomParameters()
    }
    
    private func start(appId: String) {
        // Create and config the policy //
        let policy = HSPolicy()
        policy.automaticInterceptorPolicy.interceptorType = urlRequestInterceptionType
        policy.automaticInterceptorPolicy.set(interceptedDomains: ["sample-ios.pxchk.net"], forAppId: appId)
        policy.doctorAppPolicy.enabled = true
        
        
        // Start HUMAN SDK with your AppID //
        do {
            try HumanSecurity.start(appId: appId, policy: policy)
        }
        catch {
            print("error: \(error)")
        }
    }
    
    private func setCustomParameters() {
        var customParameters = [String: String]()
        customParameters["custom_param1"] = "hello"
        customParameters["custom_param2"] = "world"
        do {
            try HumanSecurity.BD.setCustomParameters(parameters: customParameters, forAppId: pxAppId)
        }
        catch {
            print("error: \(error)")
        }
    }
}

File: Android/HUMAN_SDK_Demo/app/src/main/java/com/humansecurity/human_sdk_demo/HumanManager.kt
package com.humansecurity.human_sdk_demo

import android.app.Application
import com.humansecurity.mobile_sdk.HumanSecurity
import com.humansecurity.mobile_sdk.main.HSBotDefenderDelegate
import com.humansecurity.mobile_sdk.main.policy.HSAutomaticInterceptorType
import com.humansecurity.mobile_sdk.main.policy.HSPolicy
import com.humansecurity.mobile_sdk.main.policy.HSStorageMethod

object HumanManager: HSBotDefenderDelegate {

    // HumanDelegate

    override fun botDefenderRequestBlocked(url: String?, appId: String) {
        println("Request Blocked")
    }

    override fun botDefenderChallengeSolved(appId: String) {
        println("Challenge Solved")
    }

    override fun botDefenderChallengeCancelled(appId: String) {
        println("Challenge Cancelled")
    }

    override fun botDefenderChallengeRendered(appId: String) {
        println("Challenge Rendered")
    }

    override fun botDefenderChallengeRenderFailed(appId: String) {
        println("Challenge Render Failed")
    }

    override fun botDefenderDidUpdateHeaders(headers: HashMap<String, String>, appId: String) {
        println("Headers Were Updated")
    }

    // properties

    const val appId = "PXj9y4Q8Em"

    // HUMAN

    fun start(application: Application) {
        println("SDK version: ${HumanSecurity.sdkVersion()}")

        start(application, appId)
        setCustomParameters()
    }

    private fun start(application: Application, appId: String) {
        // Create and configure the policy //
        val policy = HSPolicy()
        policy.storageMethod = HSStorageMethod.DATA_STORE
        policy.automaticInterceptorPolicy.interceptorType = HSAutomaticInterceptorType.INTERCEPT_AND_RETRY_REQUEST
        policy.doctorAppPolicy.enabled = true

        // Start HUMAN SDK with your AppID //
        HumanSecurity.start(application, appId, policy)
    }

    private fun setCustomParameters() {
        val customParameters = HashMap<String, String>()
        customParameters["custom_param1"] = "hello"
        customParameters["custom_param2"] = "world"
        HumanSecurity.BD.setCustomParameters(customParameters, null)
    }
}
